schema-engine/sql-introspection-tests/tests/relations/mod.rs

```diff
@@ -322,6 +322,62 @@ async fn a_prisma_many_to_many_relation(api: &mut TestApi) -> TestResult {
     Ok(())
 }
 
+#[test_connector(tags(Mysql), exclude(Vitess))]
+async fn a_broken_prisma_many_to_many_relation(api: &mut TestApi) -> TestResult {
+    api.barrel()
+        .execute(|migration| {
+            migration.create_table("User", |t| {
+                t.add_column("id", types::integer().increments(true));
+                t.add_constraint("User_pkey", types::primary_constraint(["id"]));
+            });
+
+            migration.create_table("Post", |t| {
+                t.add_column("id", types::integer().increments(true));
+                t.add_constraint("Post_pkey", types::primary_constraint(["id"]));
+            });
+
+            migration.create_table("_PostToUser", |t| {
+                t.add_column("A", types::integer().nullable(false).unique(false));
+                t.add_column("B", types::integer().nullable(false).unique(false));
+
+                t.add_foreign_key(&["A"], "Post", &["id"]);
+                t.add_foreign_key(&["B"], "User", &["id"]);
+
+                t.add_index("test", types::index(vec!["A", "B"]).unique(true));
+            });
+        })
+        .await?;
+
+    api.expect_re_introspect_warnings(
+        indoc! {r##"
+        model Post {
+            id   Int    @id @default(autoincrement())
+            User Author[] @relation("PostToUser")
+        }
+
+        model Author {
+            id   Int    @id @default(autoincrement())
+            Post Post[] @relation("PostToUser")
+
+            @@map("User")
+        }
+    "##},
+        expect![[r#"
+            *** WARNING ***
+
+            These models were enriched with `@@map` information taken from the previous Prisma schema:
+              - "Author"
+
+            The following models have broken many-to-many relations, which can produce unexpected behavior. This can occur when the alphabetical ordering of tables used in relations is violated. You should ensure that the table and model names in the relation are ordered such that the first table (referred to as 'A' in the relation) is alphabetically smaller than the second table (referred to as 'B').
+              - "Author"
+              - "Post"
```

Comment:
Not sure if that's feasible but I think an error message like the following would be more actionable for me as a user:
```suggestion
            The following models have broken many-to-many relations, which can produce unexpected behavior. This can occur when the alphabetical ordering of table names does not align with the alphabetical ordering of model names. You can fix this by renaming the many-to-many relation table as suggested below:
              - "Author" <> "Post" - current: "PostToUser" => expected: "UserToPost" or "AuthorToPost"
```

Wdyt? Feasible? Clearer? (Correct?) 🤔

Reply:
The list of models is something standard we generate for every warning, but we can change it here if it's more readable. The actual action is difficult to specify precisely because the user could end up with the ordering being violated in different ways. The two main cases I found was the user using `@@map` to break the ordering or MySQL being configured to lowercase names of tables. Basically the action is that the user needs to correct their schema file so that the two produce the same ordering. It's not a good Prisma behavior that this can even happen, but if we don't fix it, we need to at least communicate it.

Reply:
I have an idea for a better error message. I can add something like this
```
Prisma uses alphabetical ordering of model names to create many-to-many relations. These two models currently produce reverse of the expected ordering: {modelA} <-> {modelB}
```

Reply:
commited the new message

Reply:
So you're saying we cannot create an error message with a direct "to fix this do X" message, because the potential cases through which this error can happen are too complex or even unknown? 🤔

Reply:
Basically the issue is that the table names when ordered by model names and the database names are different.